### PR TITLE
LaTeX state representation in Jupyter

### PR DESCRIPTION
**Pull request by minestarks**
Implement `_repr_latex_` as per IPython spec (see https://ipython.readthedocs.io/en/stable/config/integrating.html) so that Jupyter editors can provide the option of displaying the output as LaTeX .

latex

### CODE REVIEW

File: pip/qsharp/_qsharp.py [hunk 1]
```diff
@@ -343,6 +343,9 @@ def __str__(self) -> str:
     def _repr_html_(self) -> str:
         return self.__data._repr_html_()
 
+    def _repr_latex_(self) -> str:
+        return self.__data._repr_latex_()
```
Comment (DmitryVasilevsky) [inline pip/qsharp/_qsharp.py:347]: Make sure that there's something displayed when LaTeX representation isn't available.

Comment (DmitryVasilevsky) [inline pip/qsharp/_qsharp.py:347]: Make sure that there's something displayed when LaTeX representation isn't available.

**Review by DmitryVasilevsky — APPROVED** [review]

**Review by billti — APPROVED** [review]

**Review by swernli — APPROVED** [review]

**Review by idavis — APPROVED** [review]

File: npm/qsharp/src/compiler/compiler.ts [hunk 2]
```diff
@@ -309,7 +309,7 @@ export function onCompilerEvent(msg: string, eventTarget: IQscEventTarget) {
     case "DumpMachine":
       qscEvent = makeEvent("DumpMachine", {
         state: qscMsg.state,
-        stateLatex: qscMsg.stateLatex,
+        stateLatex: qscMsg.stateLatex, // can be null or undefined or something, check
```
Comment (minestarks) [inline npm/qsharp/src/compiler/compiler.ts:312]:
```suggestion
        stateLatex: qscMsg.stateLatex,
```

File: pip/qsharp/_qsharp.py (hunk 1 above)
Comment (minestarks) [reply pip/qsharp/_qsharp.py:347]: Fixed, tested and confirmed that jupyter falls back to HTML when this returns `None`. Thanks!

**Pull request merged**